src/bin/ff-analogy.rs

```diff
@@ -0,0 +1,102 @@
+use std::io::BufRead;
+
+use clap::{App, AppSettings, Arg, ArgMatches};
+use finalfusion::similarity::Analogy;
+use finalfusion_utils::{read_embeddings_view, EmbeddingFormat};
+use stdinout::{Input, OrExit};
+
+static DEFAULT_CLAP_SETTINGS: &[AppSettings] = &[
+    AppSettings::DontCollapseArgsInUsage,
+    AppSettings::UnifiedHelpMessage,
+];
+
+fn parse_args() -> ArgMatches<'static> {
+    App::new("ff-analogy")
+        .settings(DEFAULT_CLAP_SETTINGS)
+        .arg(
+            Arg::with_name("format")
+                .short("f")
+                .value_name("FORMAT")
+                .help("Embedding format: finalfusion, finalfusion_mmap, word2vec, text, or textdims (default: finalfusion)")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("neighbors")
+                .short("k")
+                .value_name("K")
+                .help("Return K nearest neighbors (default: 10)")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("EMBEDDINGS")
+                .help("Embeddings file")
+                .index(1)
+                .required(true),
+        )
+        .arg(
+            Arg::with_name("INPUT")
+            .help("Input words")
+            .index(2))
+        .get_matches()
+}
+
+struct Config {
+    embeddings_filename: String,
+    embedding_format: EmbeddingFormat,
+    k: usize,
+}
+
+fn config_from_matches<'a>(matches: &ArgMatches<'a>) -> Config {
+    let embeddings_filename = matches.value_of("EMBEDDINGS").unwrap().to_owned();
+
+    let embedding_format = matches
+        .value_of("format")
+        .map(|f| EmbeddingFormat::try_from(f).or_exit("Cannot parse embedding format", 1))
+        .unwrap_or(EmbeddingFormat::FinalFusion);
```

Comment:
This becomes `unwrap` when `default_value` is used above. The same for unwrapping `k` below.